cmd/ctl/main.go

```diff
@@ -20,20 +20,45 @@ import (
 	"context"
 	"fmt"
 	"os"
+	"runtime"
+	"strings"
+
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
 
 	ctlcmd "github.com/cert-manager/cert-manager/cmd/ctl/cmd"
 	"github.com/cert-manager/cert-manager/internal/cmd/util"
+	logf "github.com/cert-manager/cert-manager/pkg/logs"
 )
 
 func main() {
 	stopCh, exit := util.SetupExitHandler(util.AlwaysErrCode)
 	defer exit() // This function might call os.Exit, so defer last
 
+	logf.InitLogs()
+	defer logf.FlushLogs()
+
+	// In cmctl, we are using cmdutil.CheckErr, which will call os.Exit(1) if it receives an error.
+	// Instead, we want to do a soft exit, and use SetExitCode to set the correct exit code.
```

Comment:
**question:** The comment describes what the code does (i.e., use a custom exiting logic instead of CheckErr) but it doesn't explain why. The "why" seems to be the key here, and I can't quite understand why we are doing what we are doing here 😅

Reply:
The cmdutil.CheckErr function was used already in code, I think it might be because it contains some logic to "beatify" some of the errors and display more human-readable errors.